### PR TITLE
refactor decryption service

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did** Refactored the decryption service

**- How I did it** Introduced a new abstract class AtKeyDecryption. 

**- How to verify it** The encrypted value should be decrypted successfully.

**- Description for the changelog**Refactor decryption logic into decryption service.

### CODE REVIEW

File: at_client/lib/src/decryption_service/decryption_manager.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    // If sharedBy not equals currentAtSign, return SharedKeyDecryption
+    // Eg: currentAtSign is @bob and key is phone@alice
+    if (atKey.sharedBy != currentAtSign) {
+      return SharedKeyDecryption();
+    }
+    // Return SelfKeyDecryption for self keys.
+    // Eg: currentAtSign is @bob and _phone.wavi@bob (or) phone@bob (or) @bob:phone@bob
+    if ((atKey.sharedWith == null || atKey.sharedWith == currentAtSign) &&
+            atKey.sharedBy == currentAtSign ||
+        atKey.key!.startsWith('_')) {
```
Comment (gkc) [inline at_client/lib/src/decryption_service/decryption_manager.dart:20]: For readability I'd suggest either put all of `(atKey.sharedWith == null || atKey.sharedWith == currentAtSign) && atKey.sharedBy == currentAtSign` on same line of code, or wrap everything before the `||` with another set of round brackets, or both

File: at_client/lib/src/decryption_service/shared_key_decryption.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import 'package:at_client/src/response/default_response_parser.dart';
+import 'package:at_commons/at_builders.dart';
+import 'package:at_commons/at_commons.dart';
+import 'package:at_utils/at_logger.dart';
+
+/// Class responsible for decrypting the value of shared key's that are not owned
+/// by currentAtSign
+/// Example:
+/// CurrentAtSign: @bob
+/// lookup:phone@alice.
```
Comment (gkc) [inline at_client/lib/src/decryption_service/shared_key_decryption.dart]: please remove the full stop after 'alice' in the line above

**Review by gkc — APPROVED** [review]
Suggested a couple of minor changes for readability, otherwise LGTM. (Plus, of course, fix the test failure)

File: at_client/lib/src/decryption_service/decryption_manager.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/decryption_service/decryption_manager.dart:20]: Added round brackets

File: at_client/lib/src/decryption_service/shared_key_decryption.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/decryption_service/shared_key_decryption.dart]: Removed the full stop.

**Review by gkc — APPROVED** [review]
LGTM

**Pull request merged**